src/StatusQ/Controls/StatusPasswordInput.qml

```diff
@@ -0,0 +1,108 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+/*!
+   \qmltype StatusPasswordInput
+   \inherits Item
+   \inqmlmodule StatusQ.Controls
+   \since StatusQ.Controls 0.1
+   \brief The StatusPasswordInput control provides a generic user password input with an option to display signing phrase
+
+   Example of how to use it:
+
+   \qml
+        StatusPasswordInput {
+            signingPhrase: "orange hello cygnet"
+            placeholderText: qsTr("Password")
+        }
+   \endqml
+
+   For a list of available components see StatusQ.
+*/
+
+TextField {
+    id: root
+
+    /*!
+        \qmlproperty string StatusPasswordInput::signingPhrase
+        This property sets the signingPhrase of TextField field and signing seed phrase.
+    */
+    property string signingPhrase: ""
+
+    QtObject {
+        id: d
+
+        readonly property int inputTextPadding: 16
+        readonly property int pixelSize: 15
+        readonly property int radius: 8
+        readonly property int signingPhrasePadding: 8
+        readonly property int signingPhraseWordPadding: 8
+        readonly property int signingPhraseWordsSpacing: 8
+        readonly property int signingPhraseWordsHeight: 30
+
+    }
+
+    leftPadding: d.inputTextPadding
+    rightPadding: root.signingPhrase !== ""?
+                      phrase.width + phrase.anchors.leftMargin + phrase.anchors.rightMargin :
+                      d.inputTextPadding
```

Comment:
Ok, I see the implementation has changed in the meantime when I was writing this comment :sweat_smile: 

Making `TextField` a root component and doing this:
```
    RowLayout {
        id: phrase
        anchors.right: parent.right
    }
```
is not a good idea. `StatusPasswordInput` will have the wrong implicit sizes and as a consequence, it will be impossible to use it correctly in layouts. If the layout is clipped the `phrase` will become invisible.

Reply:
@osmaczko I am not sure that I follow what you want to be changed here?

> Making TextField a root component and doing this:

Have you seen how this component should look like?

> As an experiment you could try to do this:

Have you tried this?

I don't see any issues in the way it's done.

Reply:
> Have you tried this?

No, sorry  :sweat: , I just tried it now, and it works as expected within layouts

I analyzed the code and built the wrong picture in my head. My above concerns would be only true if: `anchors.left: parent.right` which is not the case. Forget about the comment :D